### PR TITLE
fix: showPasswordLabel and hidePasswordLabel not working

### LINKED ISSUE #15986: [Bug]: [TextInput.PasswordInput] showPasswordLabel and hidePasswordLabel not working
### Package

@carbon/react

### Browser

Chrome

### Package version

1.53.0

### React version

18.2.0

### Description

In TextInput.PasswordInput component `showPasswordLabel` and `hidePasswordLabel` not working. 

### Reproduction/example

https://stackblitz.com/edit/github-pe88wq?file=src%2FApp.jsx

### Steps to reproduce

https://stackblitz.com/edit/github-pe88wq?file=src%2FApp.jsx

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Application/PAL

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by preetibansalui**
Closes #15986 

ShowPasswordLabel and HidePasswordLabel props were not working.

#### Testing / Reviewing

1. Create a PasswordInput component 
2. hover on eye icon on right side of password input field.
3. It should display text we assign to prop 'showPasswordLabel'
4. Now click on that eye icon.
5. on hover it should display text we assign to prop 'hidePasswordLabel'

### CODE REVIEW

File: packages/react/src/components/TextInput/PasswordInput.tsx [hunk 1]
```diff
@@ -309,6 +311,24 @@ const PasswordInput = React.forwardRef(function PasswordInput(
...
+      align = `${tooltipPosition}-right`;
+    }
+    if (tooltipAlignment === 'start') {
+      align = `${tooltipPosition}-left`;
+    }
+  }
+
+  if (tooltipPosition === 'right' || tooltipPosition === 'left') {
+    align = tooltipPosition;
+  }
```
Comment (tay1orjones) [inline packages/react/src/components/TextInput/PasswordInput.tsx:329]: Is this necessary? Tooltip now supports new alignment options like `top-end` and `top-start`.

This code pattern is something we'll eventually want to remove from the entire codebase, I added it as a line item within 
* https://github.com/carbon-design-system/carbon/issues/16009

File: packages/react/src/components/TextInput/PasswordInput.tsx [hunk 2]
```diff
@@ -329,18 +349,18 @@ const PasswordInput = React.forwardRef(function PasswordInput(
...
-        )}
-        {passwordVisibilityIcon}
-      </button>
+      <Tooltip
+        align={align}
+        className={`${prefix}--toggle-password-tooltip`}
+        label={passwordIsVisible ? hidePasswordLabel : showPasswordLabel}>
+        <button
+          type="button"
+          className={passwordVisibilityToggleClasses}
```
Comment (tay1orjones) [inline packages/react/src/components/TextInput/PasswordInput.tsx:356]: Does this variable need to change at all? Are the classes defined in it all still needed? If not, are there some associated styles we can remove in `_text-input.scss`?

File: packages/react/src/components/TextInput/PasswordInput.tsx [hunk 3]
```diff
@@ -329,18 +349,18 @@ const PasswordInput = React.forwardRef(function PasswordInput(
...
+          disabled={disabled}
```
Comment (tay1orjones) [inline packages/react/src/components/TextInput/PasswordInput.tsx:357]: I think `Tooltip` handles this, but did you confirm the tooltip doesn't show if the button is disabled? I can't configure the storybook to disable that button.

File: packages/styles/scss/components/text-input/_text-input.scss [hunk 4]
```diff
@@ -131,6 +131,11 @@
     transition: outline $duration-fast-01 motion(standard, productive);
   }
 
+  .#{$prefix}--toggle-password-tooltip .#{$prefix}--popover {
+    position: absolute;
+    inset-inline-start: -(convert.to-rem(40px));
+  }
```
Comment (tay1orjones) [inline packages/styles/scss/components/text-input/_text-input.scss]: Does this work with all alignment options? Looking at it in the dev tools I don't see `position: absolute` modifying it in any way - could possibly be removed?

**Review by tay1orjones — COMMENTED** [review]
Functionally looks great! Just a few questions on the code

File: packages/react/src/components/TextInput/PasswordInput.tsx (hunk 3 above)
Comment (preetibansalui) [reply packages/react/src/components/TextInput/PasswordInput.tsx:357]: Yes, I checked the disabled functionality, it's not showing the tooltip if we pass disabled as true. I tried to remove this as well to check if tooltip is handling it or not, but then it was showing tooltip even disabled was true. So I think we have to pass it here.

File: packages/react/src/components/TextInput/PasswordInput.tsx (hunk 2 above)
Comment (preetibansalui) [reply packages/react/src/components/TextInput/PasswordInput.tsx:356]: These styles were already there, however I tried to check the classes and found that we are adding/removing few classes based on disabled and tooltip alignments, although few of them we can remove I think so, will remove accordingly.

File: packages/react/src/components/TextInput/PasswordInput.tsx (hunk 1 above)
Comment (preetibansalui) [reply packages/react/src/components/TextInput/PasswordInput.tsx:329]: Oh, the proptypes defined in the component were still using the old values. Do you want me to cover these changes as part of this PR or take it as a separate PR with #16009.

**Review by benjamin-kurien — CHANGES_REQUESTED** [review]
Hi Preeti, the functionality looks good. 
From a UX Perspective -  to avoid the container from bleeding off the page and hiding important information, consider restricting the container of the tooltip to be only aligned to end of the input field. [Refer this variant here](https://carbondesignsystem.com/components/tooltip/usage/#alignment)

**Comment by preetibansalui** [comment]
Hi Benjamin, Thanks for the feedback. As per the suggestion, I will apply the default functionality of 'tooltipPosition' as 'bottom' and for 'tooltipAlignment' as 'end'. 

But just to add here, user can still change it as we are exposing these 2 props to user and possible values are as below:

tooltipPosition: 'top' | 'right' | 'bottom' | 'left';
tooltipAlignment: 'start' | 'center' | 'end';

**Review by tw15egan — APPROVED** [review]

**Review by benjamin-kurien — APPROVED** [review]

**Pull request merged**